{{ cookiecutter.repo_name }}/.github/workflows/ci.yml

```diff
@@ -20,17 +20,11 @@ jobs:
         with:
           python-version: "3.11"
 
-      - name: Cache pre-commit
-        uses: actions/cache@v3
-        with:
-          path: ~/.cache/pre-commit
-          key: pre-commit-3|${{ env.pythonLocation }}|${{ hashFiles('.pre-commit-config.yaml') }}
-
-      - name: Install pre-commit
-        run: pip3 install pre-commit
+      - name: Install venv and pre-commit
+        run: make
```

Comment:
i guess we shouldn't be so brutal :D we want to use cached version to make this step much quicker and avoid setting up venv from scratch

Reply:
The problem is that we can't run make functions without creating a virtual environment and sourcing it first. Because of this, I'm not sure if it's worth continuing to work on the Makefile. Since we can't use it everywhere, it would require maintaining code in two places—both the Makefile and the ci.yml.